test/integration/src/test/java/io/pravega/test/integration/DelegationTokenTest.java

```diff
@@ -109,4 +110,109 @@ private void writeAnEvent(int tokenTtlInSeconds) throws ExecutionException, Inte
             pravegaCluster.close();
         }
     }
+
+    /**
+     * This test verifies that a event stream reader continues to read events as a result of automatic delegation token
+     * renewal, after the initial delegation token it uses expires.
+     *
+     * We use an extraordinarily high test timeout and read time outs to account for any inordinate delays that may be
+     * encountered in testing environments.
+     */
+    @Test(timeout = 50000)
+    public void testDelegationTokenGetsRenewedAfterExpiry() throws InterruptedException {
+        // Delegation token renewal threshold is 5 seconds, so we are using 6 seconds as Token TTL so that token doesn't
+        // get renewed before each use.
+        ClusterWrapper pravegaCluster = new ClusterWrapper(true, 6);
+        try {
+            pravegaCluster.initialize();
+
+            final String scope = "testscope";
+            final String streamName = "teststream";
+            final int numSegments = 1;
+
+            final ClientConfig clientConfig = ClientConfig.builder()
+                    .controllerURI(URI.create(pravegaCluster.controllerUri()))
+                    .credentials(new DefaultCredentials("1111_aaaa", "admin"))
+                    .build();
+            log.debug("Done creating client config.");
+
+            @Cleanup
+            final StreamManager streamManager = StreamManager.create(clientConfig);
+            assertNotNull(streamManager);
+            log.debug("Done creating stream manager.");
+
+            boolean isScopeCreated = streamManager.createScope(scope);
+            assertTrue("Failed to create scope", isScopeCreated);
+            log.debug("Done creating stream manager.");
+
+            boolean isStreamCreated = streamManager.createStream(scope, streamName, StreamConfiguration.builder()
+                    .scalingPolicy(ScalingPolicy.fixed(numSegments))
+                    .build());
+            Assert.assertTrue("Failed to create the stream ", isStreamCreated);
+
+            @Cleanup
+            final EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scope, clientConfig);
+
+            // Perform writes on a separate thread.
+            Runnable runnable = () -> {
+                @Cleanup
+                EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
+                        new JavaSerializer<String>(),
+                        EventWriterConfig.builder().build());
+
+                for (int i = 0; i < 10; i++) {
+                    String msg = "message: " + i;
+                    try {
+                        writer.writeEvent(msg).get();
```

Comment:
I think you might want join() instead of .get()

Reply:
Sure, that simplifies this code. I've made the change as suggested.